src/coreclr/jit/fgopt.cpp

```diff
@@ -3207,6 +3207,22 @@ bool Compiler::fgOptimizeUncondBranchToSimpleCond(BasicBlock* block, BasicBlock*
     // `fgUpdateFlowGraph` has been called with `doTailDuplication` set to true, and the
     // backend always calls `fgUpdateFlowGraph` with `doTailDuplication` set to false.
     assert(!block->IsLIR());
+    
+    // The optimization doesn't make sense in this case
+    if (target->bbWeight == 0.0)
+    {
+        return false;
+    }
```

Comment:
```suggestion
    // The optimization doesn't make sense in this case
    if (target->isRunRarely())
    {
        return false;
    }
```
Also this check is quite cheap so I'd move it up to the top of the method as the first check.